unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -507,6 +507,48 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
     }
     return res;
   }
+  
+  private class Code {
+    public final Character value;
+    public Code(Character value) {
+      this.value = value;
+    }
+    public boolean isEmpty(){
+      return value == null;
+    }
+  }
+
+  public UTF8String translate(UTF8String matchingString, UTF8String replaceString) {
+    String srcStr = this.toString();
+    String matching = matchingString.toString();
+    String replace = replaceString.toString();
+    Map<Character, Code> dict = new HashMap<>();
+    int i = 0;
+    int j = 0;
+    while (i < matching.length() && j < replace.length()) {
+      // For multiple mapping, the first one wins.
+      if (null == dict.get(matching.charAt(i))) {
```

Comment:
nit: In order to support Unicode characters that can't fit in a Char (using two Char in UTF-16), we should use code point  here. We could add a TODO for it now.